linkerd/meshtls/util/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+use linkerd_error::Result;
+use linkerd_identity::Id;
+use std::io;
+use std::str::FromStr;
+
+fn extract_ids_from_cert(cert: &[u8]) -> Result<Vec<Id>> {
+    use x509_parser::prelude::*;
+    let (_, c) = X509Certificate::from_der(cert)?;
```

Comment:
Since this changes our DNSName validation, I think we should make this change discretely from the rest of this PR. I.e.

1. A PR that adds a new `linkerd-meshtls-verifier` crate using x509_parser, without URI handling;
2. This PR.